File: bionic-js-tool/testing-code/swift/project-with-host-files/HostProject/host/TeslaRoadster.swift
import JavaScriptCore
import Bjs

class TeslaRoadster: MotorVehicle {
    
    class var `default`:TeslaRoadster? {
        get {
            return bjs.getObj(bjs.getProperty(self.bjsClass, "default"), TeslaRoadster.bjsFactory)
        }
    }
    
    var serialized:BjsAnyObject {
        get {
            return TeslaRoadster.bjs.getAny(bjsGetProperty("serialized"))
        }
    }
    
    var canTravelInTheSpace:Bool? {
        get {
            return TeslaRoadster.bjs.getBool(bjsGetProperty("canTravelInTheSpace"))
        }
    }
    
    private static var _bjsLocator: BjsLocator = BjsLocator("BeautifulVehicles", "TeslaRoadster")
    override class var bjsLocator: BjsLocator { _bjsLocator }
    override class func bjsFactory(_ jsObject: JSValue) -> TeslaRoadster { TeslaRoadster(jsObject) }
}

File: bionic-js-tool/testing-code/java/project-with-host-files/HostProject/src/motor-vehicles/java/test/project/host/TeslaRoadster.java
package test.project.host;

import jjbridge.api.runtime.JSReference;
import jjbridge.api.value.strategy.FunctionCallback;
import bionic.js.Bjs;
import bionic.js.BjsAnyObject;
import bionic.js.BjsTypeInfo;
import bionic.js.BjsObjectTypeInfo;
import bionic.js.Lambda;
import java.util.Date;
import test.project.host.libs.MotorVehicle;

@BjsTypeInfo.BjsLocation(project = "BeautifulVehicles", module = "TeslaRoadster")
public class TeslaRoadster extends MotorVehicle {
    
    protected <T extends MotorVehicle> TeslaRoadster(Class<T> type, JSReference jsObject) {
        super(type, jsObject);
    }
    
    protected <T extends MotorVehicle> TeslaRoadster(Class<T> type, JSReference[] arguments) {
        super(type, arguments);
    }
    
    public TeslaRoadster(JSReference jsObject) {
        this(TeslaRoadster.class, jsObject);
    }
    
    public static TeslaRoadster $default$() {
        return bjs.getObj(bjs.getProperty(bjsClass, "default"), TeslaRoadster.bjsFactory, TeslaRoadster.class);
    }
    
    public BjsAnyObject serialized() {
        return bjs.getAny(bjsGetProperty("serialized"));
    }
    
    public Boolean canTravelInTheSpace() {
        return bjs.getBoolean(bjsGetProperty("canTravelInTheSpace"));
    }
    
    private static final JSReference bjsClass = BjsObjectTypeInfo.get(TeslaRoadster.class).bjsClass();
    public static final Bjs bjs = BjsObjectTypeInfo.get(TeslaRoadster.class).bjsLocator.get();
    public static final Bjs.JSReferenceConverter<TeslaRoadster> bjsFactory = TeslaRoadster::new;
}